libmariadb/mariadb_lib.c

```diff
@@ -2298,20 +2447,21 @@ int mthd_my_read_query_result(MYSQL *mysql)
   {
     int error=mysql_handle_local_infile(mysql, (char *)pos, can_local_infile);
 
-    if ((length=ma_net_safe_read(mysql)) == packet_error || error)
+    if ((length=ma_net_safe_read(mysql, NULL)) == packet_error || error)
       return(-1);
     goto get_info;				/* Get info packet */
   }
   if (!(mysql->server_status & SERVER_STATUS_AUTOCOMMIT))
     mysql->server_status|= SERVER_STATUS_IN_TRANS;
 
   mysql->extra_info= net_field_length_ll(&pos); /* Maybe number of rec */
-  if (!(fields=mysql->methods->db_read_rows(mysql,(MYSQL_FIELD*) 0,
-                                            ma_result_set_rows(mysql))))
-    return(-1);
-  if (!(mysql->fields=unpack_fields(mysql, fields, &mysql->field_alloc,
-				    (uint) field_count, 1)))
-    return(-1);
+
+  if (!(mysql->fields=mthd_my_read_metadata(mysql, field_count,
+          7 + ma_extended_type_info_rows(mysql))))
```

Comment:
`ma_result_set_rows` returns 8 or 9, which worked if calling `db_read_rows`. Now that we're reading meta directly from `mthd_my_read_metadata`, by default field = 7. [ref](https://github.com/mysql/mysql-server/blob/bd9c260cf7026a54d9bf91ce6782cdb4e6a25c71/sql-common/client.c#L4990-L4991)